### PR TITLE
Add MS MARCO dataset, and modify insuranceQA to accept version (V1 or V2)

### PR DESCRIPTION
**Pull request by pcgreat**
Per discussed in #190, MS MARCO is supported, and both versions (V1/V2) of insuranceQA are supported. Tests and README are also updated

### CODE REVIEW

File: parlai/tasks/insuranceqa/agents.py [hunk 1]
```diff
@@ -4,22 +4,33 @@
...
-        opt['datafile'] = _path(opt)
+        opt['datafile'] = _path("V1", opt)
         super().__init__(opt, shared)
+
+
+# V2 InsuranceQA task
+class V2Teacher(FbDialogTeacher):
+    def __init__(self, opt, shared=None):
+        opt['datafile'] = _path("V2", opt)
+        super().__init__(opt, shared)
```
Comment (alexholdenmiller) [inline parlai/tasks/insuranceqa/agents.py]: nice!

File: parlai/tasks/insuranceqa/build.py [hunk 2]
```diff
@@ -11,99 +11,173 @@
...
+                # save good answers and candidates
+                s = '1 ' + q + '\t' + "|".join(good_ans) + '\t\t' + "|".join(good_ans + bad_ans)
+                fout.write(s + '\n')
+        fout.close()
+
+
+class ParseInsuranceQAV2(ParseInsuranceQA):
+    version = "V2"
+    label2answer_fname = "InsuranceQA.label2answer.token.encoded.gz"
+    # TODO: support 500, 1000, 1500 (# of label candidates)
```
Comment (alexholdenmiller) [inline parlai/tasks/insuranceqa/build.py]: could have the build file set up all of them, and then the teacher in the agent file decide which one to use

for example, the babi tasks have a "Task1kTeacher" and "Task10kTeacher" for the two different version, but you can specify specific tasks with "babi:Task1k:1" or "babi:task10k:2".

File: parlai/tasks/insuranceqa/build.py [hunk 3]
```diff
@@ -11,99 +11,173 @@
...
 def build(opt):
     dpath = os.path.join(opt['datapath'], 'InsuranceQA')
-    version = None
 
-    if not build_data.built(dpath, version_string=version):
+    if not build_data.built(dpath):
         print('[building data: ' + dpath + ']')
-        if build_data.built(dpath):
-            # An older version exists, so remove these outdated files.
-            build_data.remove_dir(dpath)
```
Comment (alexholdenmiller) [inline parlai/tasks/insuranceqa/build.py]: actually, these changes here just got overwritten--we updated the build functions across all tasks and would like to keep the version on the left (it allows interrupted downloads to automatically be resumed!)

File: parlai/tasks/ms_marco/agents.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+# of patent rights can be found in the PATENTS file in the same directory.
+import json
+import os
+
+from parlai.core.dialog_teacher import DialogTeacher
+from parlai.core.fbdialog_teacher import FbDialogTeacher
+from .build import build
+
+
+def _path(opt, is_passage=False, dt=''):
```
Comment (alexholdenmiller) [inline parlai/tasks/ms_marco/agents.py]: same comment as above, might as well switch dt back to not being in the args

File: parlai/tasks/ms_marco/build.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+            cands = "|".join([d["passage_text"] for d in dic["passages"]])
+        s = '1 ' + lq + '\t' + ans.lstrip("|") + '\t\t' + cands
+        fout2.write(s + '\n')
+    fout2.close()
+
+
+def build(opt):
+    dpath = os.path.join(opt['datapath'], 'MS_MARCO')
+
+    if not build_data.built(dpath):
```
Comment (alexholdenmiller) [inline parlai/tasks/ms_marco/build.py]: could you incorporate the changes to other build files here as well? we could also update it after landing this

File: tests/test_downloads.py [hunk 6]
```diff
@@ -379,6 +379,44 @@ def test_coco_datasets(self):
...
+            teacher = DefaultTeacher(opt)
+            reply = teacher.act()
+            check(opt, reply)
+
+            teacher = PassageTeacher(opt)
+            reply = teacher.act()
+            check(opt, reply)
+
+        shutil.rmtree(self.TMP_PATH)
+
```
Comment (alexholdenmiller) [inline tests/test_downloads.py]: thanks for adding these!

**Comment by alexholdenmiller** [comment]
also can we move the whole directory ms_marco => msmarco? just makes it a little easier to type on the command line

File: parlai/tasks/task_list.py [hunk 7]
```diff
@@ -219,5 +219,12 @@
         "task": "insuranceqa",
         "tags": [ "All",  "QA" ],
         "description": "Task which requires agents to identify high quality answers composed by professionals with deep domain knowledge. Link: https://github.com/shuzi/insuranceQA"
+    },
+    {
+        "id": "MS_MARCO",
+        "display_name": "MS_MARCO",
+        "task": "ms_marco",
```
Comment (alexholdenmiller) [inline parlai/tasks/task_list.py]: can we remove the underscore from these?

**Comment by alexholdenmiller** [comment]
thanks a ton for taking the time to add this, we really appreciate it!

**Comment by alexholdenmiller** [comment]
reverted a few comments, we can update those things later

**Comment by pcgreat** [comment]
@alexholdenmiller thanks for the instructions! I updated the pull request with those changes, and please let me know if you have any question.

File: parlai/tasks/ms_marco/agents.py [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+                d_example = json.loads(jline)
+                context = [d["passage_text"] for d in d_example['passages']]
+                question = d_example['query']
+                if self.datatype != "test":
+                    answers = d_example['answers']
+                    if not answers:
+                        answers = ["NULL"]  # empty list of answers will cause exception
+                else:
+                    answers = ["NULL"]
+                yield ("|".join(context) + '\n' + question, answers), True
```
Comment (alexholdenmiller) [inline parlai/tasks/ms_marco/agents.py]: I edited this directly, changing '|' => '\n'

**Pull request merged**